content/codespaces/getting-started-with-codespaces/getting-started-with-your-python-project-in-codespaces.md

```diff
@@ -0,0 +1,222 @@
+
+When you create a codespace, your project is created on a remote VM that is dedicated to you. By default, the container for your codespace has many languages and runtimes including Node.js, JavaScript, Typescript, nvm, npm, and yarn. It also includes a common set of tools like git, wget, rsync, openssh, and nano. 
+
+You can customize your codespace by adjusting the amount of vCPUs and RAM], [adding dotfiles to personalize your environment](/codespaces/setting-up-your-codespace/personalizing-codespaces-for-your-account), or by modifying the tools and scripts installed. 
+
+{% data variables.product.prodname_codespaces %} uses a file called `devcontainer.json` to store configurations. On launch {% data variables.product.prodname_codespaces %} uses the file to install any tools, dependencies, or other set up that might be needed for the project. For more information, see "[Configuring Codespaces for your project](/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project)."
+
+The next section shows you how to modify your tools by adding a dev container.
+
+### Step 2: Add a dev container to your codespace from a template  
+
+The default codespaces container will support running Python projects like [python-docs-hello-world](https://github.com/2percentsilk/python-docs-hello-world) out of the box. By setting up a custom container you can customize the tools and scripts that run as part of codespace creation and ensure a fully reproducible environment for all {% data variables.product.prodname_codespaces %} users in your repository.
+
+To set up your project with a custom container, you will need to use a `devcontainer.json` file to define the environment. In {% data variables.product.prodname_codespaces %} you can add this either from a template or you can create your own. For more information on dev containers, see [Configuring your codespace](/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project). 
+
+This example guides you through adding a `devcontainer.json` file from a template.
+
+1. Access the command palette (`shift command P` / `shift control P`), then start typing "dev container". Click **Codespaces: Add Development Container Configuration Files...**
+  !["Codespaces: Add Development Container Configuration Files..." in the command palette](/assets/images/help/codespaces/add-prebuilt-container-command.png)
+3. For this example, click **Python 3**. In practice, you could select any container that’s specific to Node or a combination of tools such as Python 3 and PostgresSQL.
+  ![Select Python option from the list](/assets/images/help/codespaces/add-python-prebuilt-container.png)
+4. Click the recommended version of Python.
+  ![Python version selection](/assets/images/help/codespaces/add-python-version.png)
+5. Accept the default option to add Node.js to your customization.
+  ![Add Node.js selection](/assets/images/help/codespaces/add-nodejs-selection.png)
```

Comment:
Perhaps the images above don't show up because we're referencing a combination of Node and Python assets?